homeassistant/auth/providers/homeassistant.py

```diff
@@ -106,44 +107,49 @@ async def async_load(self) -> None:
         if (data := await self._store.async_load()) is None:
             data = cast(dict[str, list[dict[str, str]]], {"users": []})
 
-        seen: set[str] = set()
+        self._async_check_for_not_normalized_usernames(data)
+        self._data = data
+
+    @callback
+    def _async_check_for_not_normalized_usernames(
+        self, data: dict[str, list[dict[str, str]]]
+    ) -> None:
+        not_normalized_usernames: set[str] = set()
 
         for user in data["users"]:
             username = user["username"]
 
-            # check if we have duplicates
-            if (folded := username.casefold()) in seen:
-                self.is_legacy = True
-
+            if self.normalize_username(username, force_normalize=True) != username:
                 logging.getLogger(__name__).warning(
                     (
                         "Home Assistant auth provider is running in legacy mode "
-                        "because we detected usernames that are case-insensitive"
-                        "equivalent. Please change the username: '%s'."
+                        "because we detected usernames that are normalized (lowercase and without spaces)."
```

Comment:
This line became longer than 88 characters. Please reformat to keep the lines below that length.